packages/cli/src/commands/storybook.js

```diff
@@ -77,21 +92,56 @@ export const handler = ({
     require.resolve('@redwoodjs/testing/config/storybook/main.js')
   )
 
-  execa(
-    `yarn ${build ? 'build' : 'start'}-storybook`,
-    [
-      `--config-dir "${storybookConfig}"`,
-      !build && `--port ${port}`,
-      !build && '--no-version-updates',
-      !managerCache && '--no-manager-cache',
-      build && `--output-dir "${buildDirectory}"`,
-      !open && !smokeTest && `--ci`,
-      smokeTest && `--ci --smoke-test`,
-    ].filter(Boolean),
-    {
-      stdio: 'inherit',
-      shell: true,
-      cwd,
+  try {
+    if (build) {
+      execa(
+        `yarn build-storybook`,
+        [
+          `--config-dir "${storybookConfig}"`,
+          `--output-dir "${buildDirectory}"`,
+          !managerCache && `--no-manager-cache`,
+        ].filter(Boolean),
+        {
+          stdio: 'inherit',
+          shell: true,
+          cwd,
+        }
+      )
+    } else if (smokeTest) {
+      execa(
+        `yarn start-storybook`,
+        [
+          `--config-dir "${storybookConfig}"`,
+          `--port ${port}`,
+          `--smoke-test`,
+          `--ci`,
+          `--no-version-updates`,
+        ].filter(Boolean),
+        {
+          stdio: 'inherit',
+          shell: true,
+          cwd,
+        }
+      )
+    } else {
+      execa(
+        `yarn start-storybook`,
+        [
+          `--config-dir "${storybookConfig}"`,
+          `--port ${port}`,
+          !managerCache && `--no-manager-cache`,
+          `--no-version-updates`,
```

Comment:
re: Line 72-37:
```
  // eslint-disable-next-line no-unused-vars
   open,
```

We do want to use the var `open` here when building to run Storybook. But because the option is negative, you'll want to add something like `!open && --no-open` (do confirm I have that correct). You can see options via `yarn start-storybook --help` in a test project.